catboost/libs/model/coreml_helpers.cpp

```diff
@@ -168,15 +321,31 @@ void ProcessOneTree(const TVector<const TreeEnsembleParameters::TreeNode*>& tree
             CB_ENSURE(node->truechildnodeid() < nodeId);
             CB_ENSURE(nodeLayerIds[node->falsechildnodeid()] == nodeLayerIds[node->truechildnodeid()]);
             nodeLayerIds[nodeId] = nodeLayerIds[node->falsechildnodeid()] + 1;
-            if (splits->ysize() <= nodeLayerIds[nodeId]) {
-                splits->resize(nodeLayerIds[nodeId] + 1);
-                auto& floatSplit = splits->at(nodeLayerIds[nodeId]);
-                floatSplit.FloatFeature = node->branchfeatureindex();
-                floatSplit.Split = node->branchfeaturevalue();
+
+            if (node->nodebehavior() == TreeEnsembleParameters::TreeNode::BranchOnValueGreaterThan) {
+                if (splits->ysize() <= nodeLayerIds[nodeId]) {
+                    splits->resize(nodeLayerIds[nodeId] + 1);
+                    auto& split = splits->at(nodeLayerIds[nodeId]);
+                    split.FloatFeature.FloatFeature = node->branchfeatureindex();
+                    split.FloatFeature.Split = node->branchfeaturevalue();
+                    split.Type = ESplitType::FloatFeature;
+                } else {
+                    auto& floatSplit = splits->at(nodeLayerIds[nodeId]).FloatFeature;
+                    CB_ENSURE(floatSplit.FloatFeature == (int)node->branchfeatureindex());
+                    CB_ENSURE(floatSplit.Split == node->branchfeaturevalue());
+                }
             } else {
-                auto& floatSplit = splits->at(nodeLayerIds[nodeId]);
-                CB_ENSURE(floatSplit.FloatFeature == (int)node->branchfeatureindex());
-                CB_ENSURE(floatSplit.Split == node->branchfeaturevalue());
+                if (splits->ysize() <= nodeLayerIds[nodeId]) {
+                    splits->resize(nodeLayerIds[nodeId] + 1);
+                    auto& split = splits->at(nodeLayerIds[nodeId]);
+                    split.OneHotFeature.CatFeatureIdx = node->branchfeatureindex();
+                    split.OneHotFeature.Value = (int)node->branchfeaturevalue();
```

Comment:
Please, add a comment that Value is not hashed value as it should be and the field is used temporarily as an index in the OneHotFeature Values array with hashed values.